### PR TITLE
feat: add debug logging of repo sub processes

### PR DESCRIPTION
**Pull request by rickeylev**
Debugging what the repo rules are up to can be difficult because Bazel doesn't provide many facilities to inspect what they're up to. This adds an environment variable, `RULES_PYTHON_REPO_DEBUG`, that, when set, will make our repo rules print out detailed information about the subprocesses they are running.

This also makes failed commands dump much more comprehensive information.

This was driven by the recent report of failures on Windows during a repo rule.

### CODE REVIEW

File: python/pip_install/pip_repository.bzl [hunk 1]
```diff
@@ -719,28 +726,14 @@ def _whl_library_impl(rctx):
     # Manually construct the PYTHONPATH since we cannot use the toolchain here
     environment = _create_repository_execution_environment(rctx, python_interpreter)
 
-    result = rctx.execute(
-        args,
+    repo_utils.execute_checked(
+        rctx,
+        op = "whl_library.ResolveRequirement({}, {})".format(rctx.attr.name, rctx.attr.requirement),
```
Comment (aignas) [inline python/pip_install/pip_repository.bzl:731]: nit: we could also have this semantics for the op, where the first item is the function name and the remaining are parameters which will be formated to: `op[0] + "(" + ",".join(op[1:]) + ")"`
```suggestion
        op = ["whl_library.ResolveRequirement", rctx.attr.name, rctx.attr.requirement],
```

**Review by aignas — APPROVED** [review]
Thanks, this is really great!

File: python/pip_install/pip_repository.bzl (hunk 1 above)
Comment (rickeylev) [reply python/pip_install/pip_repository.bzl:731]: I like the idea. It also gave me another idea: automatically add the "whl_library" part (the repo rule name) by looking for a `_repo_name` attribute we set on all our repo rules.

But, I'm strapped for time today, so I'm going to merge this as-is.

**Review by rickeylev — COMMENTED** [review]
I tried writing some tests for this but it started to spiral out of control pretty quickly (I tried mocking rctx, fail, and print 😵‍💫) and didn't test very well. I manually ran `bazel build ...` with the env set and forcing failures instead.

**Pull request merged**